packages/terra-application-layout/src/menu/_ApplicationMenu.jsx

```diff
@@ -0,0 +1,179 @@
+
+const propTypes = {
+  /**
+   * The AppDelegate instance that will be propagated to the components presented within the NavigationLayout.
+   */
+  app: AppDelegate.propType,
+  /**
+   * The element to be placed within the fill flex styled content area.
+   */
+  content: PropTypes.element,
+  /**
+   * The element to be placed within the fit top area for extensions within the layout.
+   */
+  extensions: PropTypes.element,
+  /**
+   * The Object of layout-related APIs provided to the components of the Layout.
+   */
+  layoutConfig: ApplicationLayoutPropTypes.layoutConfigPropType.isRequired,
+  /**
+   * Configuration values for the ApplicationName component.
+   */
+  nameConfig: ApplicationLayoutPropTypes.nameConfigPropType,
+  /**
+   * Delegate prop that is provided by the NavigationLayout.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType.isRequired,
+  /**
+   * The set of routes currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutRoutes: PropTypes.arrayOf(processedRoutesPropType),
+  /**
+   * The window size currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutSize: PropTypes.string,
+  /**
+   * Configuration to be provided to the ApplicationUtility component.
+   */
+  utilityConfig: ApplicationLayoutPropTypes.utilityConfigPropType,
+};
+
+class ApplicationMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.renderHeader = this.renderHeader.bind(this);
+    this.renderExtensions = this.renderExtensions.bind(this);
+    this.renderFooter = this.renderFooter.bind(this);
+    this.handleUtilityDiscloseRequest = this.handleUtilityDiscloseRequest.bind(this);
+    this.handleUtilityOnChange = this.handleUtilityOnChange.bind(this);
+  }
+
+  handleUtilityDiscloseRequest(utilityMenu) {
+    const { app } = this.props;
+
+    if (app && utilityMenu) {
+      app.disclose({
+        preferredType: modalDisclosureType,
+        size: availableDisclosureSizes.SMALL,
+        content: {
+          component: <UtilityMenuWrapper>{utilityMenu}</UtilityMenuWrapper>,
+          key: 'application-menu-utility-menu',
+        },
+      });
+    }
+  }
+
+  handleUtilityOnChange(event, key) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, key, app && app.disclose);
+  }
+
+  renderHeader(isCompact) {
+    const { nameConfig } = this.props;
+
+    if (isCompact && (nameConfig.accessory || nameConfig.title)) {
+      return (
+        <div className={cx(['menu-header'])}>
+          <ApplicationMenuName accessory={nameConfig.accessory} title={nameConfig.title} />
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderExtensions(isCompact) {
+    const { app, layoutConfig, extensions } = this.props;
+
+    if (isCompact && extensions) {
+      return React.cloneElement(extensions, { app, layoutConfig });
```

Comment:
Why do we need to pass the `app` and `layoutConfig` to the provided extensions element? If we are doing this, we should document this behavior in the props description or consumers will be confused as to by their element is passed these props.

Reply:
Also, do we need to check app before passing it? We checked it above in `handleUtilityDiscloseRequest`

Reply:
Above, we check it before we call it. Here, if it's undefined, it will just pass through cleanly.

It is documented in the readme but not on the inline prop comment. I will update that.